subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/ClientProvidedBuildActionRunner.java

```diff
@@ -44,22 +43,14 @@ public void run(BuildAction action, final BuildController buildController) {
         }
 
         final GradleInternal gradle = buildController.getGradle();
+        gradle.getStartParameter().setConfigureOnDemand(false);
```

Comment:
Now, you might ask "Why is there a 'run tasks only'" special case in the one class, but not the other? That's because of some unfortunate history, where the `BuildModelActionRunner` is the backend serving both the `getModel()` and the `newBuild()` methods. In the `newBuild()` case, we want to benefit from configure-on-demand, because we know that the user is not requesting any models.